### DISCUSSION TITLE
How to subscribe or receive typing notifications?

### DISCUSSION THREAD

**Discussion by GaoGorden** [tulir/whatsmeow#681] [General]
Hello guys,

I did not found any way to subscribe/receive "typing" notifications from others.

![image](https://github.com/user-attachments/assets/5ab04bae-aefe-46fe-afd9-69f3e85d7446)

Is there a way?

thanks!

**Comment by ghost** [comment]
You should send set "available" first on your whatsmeow client.

```go
client.SendPresence(types.PresenceAvailable)
```

**Reply by GaoGorden** [reply]
thanks for your reply.

terminal info "Marked self as available" indicate that i had updated my state, then i inputed command "subscribepresence jid", still didn't received any "typing" notifications when contacts were typing .

terminal output like below:
`...`
`[Client INFO] Successfully authenticated`
`[Main INFO] Marked self as available`
`subscribepresence 447743***@s.whatsapp.net`
`[Main INFO] online: 447743***@s.whatsapp.net`

**Reply by ghost** [reply]
Make sure both clients has marked as available. Then the server will send the `typing/compose` or `recording` notification to the target chat/client.

NB : You can't see your own typing state.

**Reply by GaoGorden** [reply]
Sorry for the late reply.

I set both accounts to Available in the Android test device, and both devices can see each other's Typing status, but I still can't see the Typing notification in the whatsmeow client.

How can I reproduce the test process you described? Thank you!

**Reply by ghost** [reply]
> You should send set "available" first on your whatsmeow client.
> 
> ```go
> client.SendPresence(types.PresenceAvailable)
> ```

I just send presence available and typing event should be received in whatsmeow client.

```go
*events.ChatPresence

```

**Reply by GaoGorden** [reply]
Thank you for your reply

Unfortunately, after calling `client.SendPresence(types.PresenceAvailable)` in the whatsmeow client, I still did not receive the Typing notification.

I am using the latest submitted version of whatsmeow.
